V8 review changes 1

kubalewski opened the pull request.

kubalewski committed:
net: ynl: fix NLA_POLICY_MAX on enums with value

Policy defines max as length of enum list, if enum attribute is
defined with a value, generated netlink policy max value is wrong.
Calculate proper max value of policy for enums that are given a value.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
dpll: spec: remove unspec attributes

Attributes are not used, remove them and start with a value given.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
dpll: spec: add 10 KHZ and 77,5 KHZ frequencies

Add missing frequencies to the dpll ynl spec, so their defines are
properly generated.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
dpll: spec: fix documentation

- remove "no holdover available" from freerun mode description
- improve description of holdover lock-status
- fix temperature typo

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
dpll: core: remove unspec attributes

Remove unspec attributes after removing from dpll netlink spec.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
ice: dpll: remove unspec attributes

Remove unspec attributes after removing from dpll netlink spec.

Fixes: 6348a829aeb3 ("ice: add admin commands to access cgu configuration")
Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
ice: dpll: remove unspec attributes

Remove unspec attributes after removing from dpll netlink spec.

Fixes: bff85d94a251 ("ice: implement dpll interface to control cgu")
Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
ice: dpll: do not store set flags

ice_dpll_pin flags field is intended to store flags that were received
after getting pin info from firmware, stop using it on set commands,
as the flags of set commands can have different meaning and values.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
ice: dpll: fix use pin state

Use ice_dpll_pin's 'state' field to provide state to the caller,
as well as to check if state change requested shall be proceeded.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
dpll: spec: remove values from subset attributes

Remove attribute values of subset attributes, they are no longer needed
as ynl lib was fixed.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
dpll: spec: reorder dpll attributes

Move device 'nest' to the end of the dpll netlink attributes.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
dpll: spec: fix device nested attribute

'device' nested attribute shall be used only for receiving data
from pin-get do/dump commands. Use it this way and define list
of expected list of attributes for device-get which are not nested.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
dpll: docs: fix DPLL_MODE_MANUAL docs

Previously the docs reffered to DPLL_MODE_FORCED mode, but its name was
changed to DPLL_MODE_MANUAL, fix it.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
dpll: spec: temperature with 3 digit float precision

Previously temperature could be supplied with one digit float precision,
use divider value of 1000 instead of 10 and allow three digit float
precision for users.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
dpll: spec: remove enum from subset definition

Enum in subset attribute definition is redundant and not used for
anything in YNL spec, remove it.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
dpll: spec: move pin-direction to pin-dpll tuple

Previously it was possible to have different pin-direction returned to
the user depending on which dpll was given as argument for pin-get cmd.
If pin was registered with multiple dplls each could have reported
different direction. I.e. driver exposes chained pins, where one pin is
an input for one dpll and an output for second dpll.
Fix it by enclosing pin-direction in the 'device' nested attribute.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
dpll: core: move pin-direction to pin-dpll tuple

Previously it was possible to have different pin-direction returned to
the user depending on which dpll was given as argument for pin-get cmd.
If pin was registered with multiple dplls each could have reported
different direction. I.e. driver exposes chained pins, where one pin is
an input for one dpll and an output for second dpll.
Fix it by enclosing pin-direction in the 'device' nested attribute.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
ice: remove include of uapi header

The header is already included in linux/dpll.h. Remove it.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
ice: remove redundant check against null arguments

Caller shall take care of checking if arguments are valid, remove it.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
ice: propagate lock function return value to the caller

Propagate error return value of netlink callback mutex lock function to
the caller, a dpll subsystem.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
ice: fix use switch case instead of if

Replace if statements with enum values to the switch case statements.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
ice: fix dev_dbg to dev_err for error cases

In case of runtime errors the dev_err macro shall be used for printing
the traces.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
ice: fix dpll/pin lookup on dpll subsytem callbacks

Add pf structure pointer to ice_dpll and ice_dpll_pin structures.
Use ice_dpll and ice_dpll_pin structure pointers as private data when
registering within dpll subsystem.
New private data does not need to perform any lookups to perform
callback requests.
Remove ice_pind_pin as it is unused.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
ice: fix extack of dpll subsystem callbacks

Fixes to dpll subsystem netlink callback extack.
Remove return values as they are already propagated.
Add info with priority range in case requested priority is out of
range.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
ice: remove double negation and variable cast

Double negation and a cast are not needed, remove it.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
ice: simplify ice_dpll_pin_state_set function

Decide expected state before calling the ice_dpll_pin_state_set
function, which simplifies the code.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
dpll: core: pass parent_priv on state_on_pin_<get/set>

Pass parent's priv data pointer when invoked callbacks are
related to the pin on pin relation.
This simplifies implementation on driver implementing a callback as it
has direct access to the parent priv, it does not have to look for
correct parent pin in its internal structures anymore.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
ice: pass parent_priv on state_on_pin_<get/set>

Implement changes on state_on_pin_<get/set> in ice driver.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
ice: remove parent hw_idx lookup

Instead of parent hw_idx lookup use an equation to obtain it.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
ice: fix use const qualifier for dpll/dpll_pin ops

Definition of callback arrays now const as expected.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
ice: fix IS_ERR macros usage in ice_dpll

Previously IS_ERR_OR_NULL was used worngly as the function cannot return
NULL, use IS_ERR instead.
In case of error, return original error value.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
ice: add notify previous source state change

When source pin changes previous source state notification shall be
also requested, request it.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
ice: fix mutex locking on releasing pins

Simplify code by locking dpll subsystem once and releasing all the
resources, instead of locking per each resource release.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
ice: use '|=' instead of '+=' when modifing capabilities field

Use proper way of setting bits for capabilities field.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
ice: rename ice_dpll_register_pins function

Align function name with dpll counterpart (rename from
ice_dpll_register_pins to ice_dpll_init_pins), similarly old
ice_dpll_init_pins function renamed to ice_dpll_init_pins_info.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
ice: simplify clock_id generation

Change the clock_id function to return clock ID on the stack instead
of using an output variable, in order to simplify the function.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
dpll: spec: remove DPLL_MODE_NCO

Remove the mode as it is not used by any of the in-kernel drivers
implementing the dpll subsystem.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
dpll: docs: remove DPLL_MODE_NCO

Remove the mode as it is not used by any of the in-kernel drivers
implementing the dpll subsystem.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
dpll: spec: remove DPLL_LOCK_STATUS_CALIBRATING

DPLL_LOCK_STATUS_CALIBRATING is removed, when dpll is locked with
a source signal it shall return DPLL_LOCK_STATUS_LOCKED, if it is
capable of holdover with that signal it shall return it's state as:
DPLL_LOCK_STATUS_LOCKED_HO_ACQ

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
dpll: docs: remove DPLL_LOCK_STATUS_CALIBRATING

DPLL_LOCK_STATUS_CALIBRATING is removed, when dpll is locked with
a source signal it shall return DPLL_LOCK_STATUS_LOCKED, if it is
capable of holdover with that signal it shall return it's state as:
DPLL_LOCK_STATUS_LOCKED_HO_ACQ

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
ice: remove DPLL_LOCK_STATUS_CALIBRATING

DPLL_LOCK_STATUS_CALIBRATING was removed, return:
DPLL_LOCK_STATUS_LOCKED - if dpll was locked
DPLL_LOCK_STATUS_LOCKED_HO_ACQ - if dpll was locked and holdover is
acquired.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
ice: propagate and use dpll_priv

Propagate and use dpll_priv to obtain pf pointer in corresponding
functions.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
ice: remove null check for pf pointer

Pointer is given back from dpll subsystem, it cannot null unless dpll
subsystem is broken.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
dpll: spec: fix notification definition

Define notifications as commands as expected by yaml spec.
Add separated notification for pin events.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
dpll: core: fix notification definition

Align with new notification names.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

jpirko reviewed (commented):
I don't follow why do you remove "pin-frequency" in "dpll: spec: move pin-direction to pin-dpll tuple"

kubalewski commented:
> I don't follow why do you remove "pin-frequency" in "dpll: spec: move pin-direction to pin-dpll tuple"

Thanks for catching this up, fixed.

kubalewski committed:
ice: expose pf related pin to corresponding netdevice

Register a recoered clock pin of a pf in its netdevice struct.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
dpll: docs: add grepability

Use full names of commands, attributes and values of dpll subsystem in
the documentation.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
dpll: fix notification scheme

Use separated notifications for dpll pins and devices.
Format the messages as corresponding get netlink commands.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
ice: fix notification scheme

Align with fixed notification scheme in dpll subsystem.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>

kubalewski committed:
mlx5: fix notification scheme

Align with fixed notification scheme in dpll subsystem.

Signed-off-by: Arkadiusz Kubalewski <arkadiusz.kubalewski@intel.com>
